lib/medtronic/medtronicSimulator.js

```diff
@@ -103,7 +103,7 @@ exports.make = function(config){
               break;
             }
 
-            if (common.computeMillisInCurrentDay(cancelled) < currSchedule[i].start ) {
+            if (common.computeMillisInCurrentDay(cancelled) <= currSchedule[i].start ) {
```

Comment:
@pazaan I'm adding one more change to this PR. What happened here is that if a temp basal is cancelled at the exact same time that a new basal schedule is starting, this if statement would not be triggered,  hence the adding of `=`. I'm merging already, but ping me if there any concerns.